tests/test_scan.py

```diff
@@ -0,0 +1,589 @@
+        return S_t, Y_t
+
+    (S_rv, Y_rv), scan_updates = aesara.scan(
+        fn=scan_fn,
+        sequences=[mus_tt, sigmas_tt],
+        non_sequences=[Gamma_rv, rng],
+        outputs_info=[{"initial": S_0_rv, "taps": [-1]}, {}],
+        strict=True,
+        name="scan_rv",
+    )
+    Y_rv.name = "Y_rv"
+
+    scan_op = Y_rv.owner.op
+    scan_args = ScanArgs.from_node(Y_rv.owner)
+
+    Gamma_in = scan_args.inner_in_non_seqs[0]
+    Y_t = scan_args.inner_out_nit_sot[0]
+    mus_t = scan_args.inner_in_seqs[0]
+    sigmas_t = scan_args.inner_in_seqs[1]
+    S_t = scan_args.inner_out_sit_sot[0]
+    rng_in = scan_args.inner_in_non_seqs[0]
+
+    mus_in = Y_rv.owner.inputs[1]
+    mus_in.name = "mus_in"
+    sigmas_in = Y_rv.owner.inputs[2]
+    N_tt = at.iscalar("N")
+    N_tt.tag.test_value = 10
+    M_tt = at.iscalar("M")
+    M_tt.tag.test_value = 2
+
+    mus_tt = at.matrix("mus")
+    mus_tt.tag.test_value = np.stack(
+        [np.arange(0.0, 10), np.arange(0.0, -10, -1)], axis=-1
+    ).astype(aesara.config.floatX)
+
+    sigmas_tt = at.ones((N_tt,))
+    sigmas_tt.name = "sigmas"
+
+    pi_0_rv = at.random.dirichlet(at.ones((M_tt,)), rng=rng, name="pi_0")
+    Gamma_rv = at.random.dirichlet(at.ones((M_tt, M_tt)), rng=rng, name="Gamma")
+
+    S_0_rv = at.random.categorical(pi_0_rv, name="S_0")
+
+    def scan_fn(mus_t, sigma_t, S_tm1, Gamma_t, rng):
+        S_t = at.random.categorical(Gamma_t[S_tm1], rng=rng, name="S_t")
+        Y_t = at.random.normal(mus_t[S_t], sigma_t, rng=rng, name="Y_t")
+        return S_t, Y_t
+
+    (S_rv, Y_rv), scan_updates = aesara.scan(
+        fn=scan_fn,
+        sequences=[mus_tt, sigmas_tt],
+        non_sequences=[Gamma_rv, rng],
+        outputs_info=[{"initial": S_0_rv, "taps": [-1]}, {}],
+        strict=True,
+        name="scan_rv",
+    )
+    Y_rv.name = "Y_rv"
+
+    scan_op = Y_rv.owner.op
+    scan_args = ScanArgs.from_node(Y_rv.owner)
+
+    Gamma_in = scan_args.inner_in_non_seqs[0]
+    Y_t = scan_args.inner_out_nit_sot[0]
+    mus_t = scan_args.inner_in_seqs[0]
+    sigmas_t = scan_args.inner_in_seqs[1]
+    S_t = scan_args.inner_out_sit_sot[0]
+    rng_in = scan_args.inner_in_non_seqs[0]
+
+    mus_in = Y_rv.owner.inputs[1]
+    mus_in.name = "mus_in"
+    sigmas_in = Y_rv.owner.inputs[2]
+    sigmas_in.name = "sigmas_in"
+
+    # The output `S_rv` is really `S_rv[1:]`, so we have to extract the actual
+    # `Scan` output: `S_rv`.
+    S_in = S_rv.owner.inputs[0]
+    S_in.name = "S_in"
+
+    return locals()
+
+
+def test_measurable_scan():
+    hmm_model_env = create_test_hmm()
+    Y_rv = hmm_model_env["Y_rv"]
+
+    scan_rv = Y_rv.owner
+    assert isinstance(scan_rv.op, Scan)
+
+    m_S_rv = MeasurableScan.create_node(scan_rv, scan_rv.outputs)
+
+    assert m_S_rv.op.inputs == scan_rv.op.inputs
+    assert m_S_rv.op.outputs == scan_rv.op.outputs
+    assert m_S_rv.inputs == scan_rv.inputs
+    # Check clones
+    assert m_S_rv.outputs == scan_rv.outputs
```

Comment:
I don't believe this should pass, because the output `TensorVariable`s use identity (e.g. `is`) comparisons.